### PR TITLE
Implement the sign-feature

### PR DESCRIPTION
**Pull request by elioschmutz**
This PR implements the signing feature.

The feature extends the newly introduced sign-transitions from #8034 with additional transition extenders which are responsible for starting the sign-process and uploading a new, signed document version.

The signing-flow looks as follow:

Bildschirmfoto 2024-09-12 um 09 41 40

ℹ️ the coral-parts are implemented by this PR.

For [TI-659]

## Checklist

_Everything has to be done/checked. Checked but not present means the author deemed it unnecessary._

- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

_Only applicable should be left and checked._

- Upgrade-Steps:
  - [ ] SQL Operations do not use imported model (see [docs](https://4teamwork.atlassian.net/wiki/spaces/4TEAM/pages/994344994/Upgrade-Steps))
  - [ ] Make it deferrable if possible
  - [ ] Execute as much as possible conditionally
  - DB-Schema migration
    - [ ] All changes on a model (columns, etc) are included in a DB-schema migration.
    - [ ] Constraint names are shorter than 30 characters (`Oracle`)
- API change:
  - [ ] Documentation is updated
  - [ ] API Changelog entry (see [guide](https://4teamwork.atlassian.net/wiki/spaces/4TEAM/pages/451248812/API+Changelog+Guidelines))
  - If breaking:
    - [ ] api-change label added
    - [ ] #delivery channel notified about breaking change
    - [ ] Scrum master is informed
- Bug fixed:
  - [ ] Resolved any Sentry issues caused by this bug
- New functionality:
  - [ ] for `document` also works for `mail`
  - [ ] for `task` also works for `forwarding`
- Further improvements needed:
  - [ ] Create follow-up stories and link them in the PR and Jira issue
- [ ] Change could impact client installations, client policies need to be adapted
- New translations
  - [ ] All msg-strings are unicode
- Change in schema definition:
  - [ ] If `missing_value` is specified, then `default` has to be set to the same value

[TI-659]: https://4teamwork.atlassian.net/browse/TI-659?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: opengever/document/transition.py [hunk 1]
```diff
@@ -15,3 +25,93 @@ def after_transition_hook(self, transition, disable_sync, transition_params):
...
         """
         self.context.finalizer = api.user.get_current().getId()
+
+
+@implementer(ITransitionExtender)
+@adapter(IDocumentSchema, IBrowserRequest)
+class DocumentFinalSigningTransitionExtender(TransitionExtender):
+
+    def after_transition_hook(self, *args, **kwargs):
+        """Start sing process
```
Comment (lukasgraf) [inline opengever/document/transition.py:35]: sing -> signing

File: opengever/api/configure.zcml [hunk 2]
```diff
@@ -823,6 +823,15 @@
       permission="zope2.View"
       />
 
+  <plone:service
```
Comment (lukasgraf) [inline opengever/api/configure.zcml:826]: Commit message: singing -> signing

**Review by lukasgraf — APPROVED** [review]
LGTM 👍! Just a few translatinos :P

File: opengever/api/configure.zcml (hunk 2 above)
Comment (elioschmutz) [reply opengever/api/configure.zcml:826]: 🎤

**Pull request merged**